### PR TITLE
Auth.authorise: store TokenParams and AuthOptions

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: ably-ios/ARTAuth.m [hunk 1]
```diff
@@ -84,8 +86,26 @@ - (ARTAuthOptions *)mergeOptions:(ARTAuthOptions *)customOptions {
     return customOptions ? [self.options mergeWith:customOptions] : self.options;
 }
 
+- (void)storeOptions:(ARTAuthOptions *)customOptions {
```
Comment (tcard) [inline ably-ios/ARTAuth.m]: If ARTAuthOptions some day adds new properties, I think chances are this method will be left outdated. Maybe add this functionality to ARTAuthOptions, as described [here](https://developer.apple.com/library/ios/documentation/General/Conceptual/DevPedia-CocoaCore/ObjectCopying.html)?

Why do we need to copy the options anyway?

Comment (ricardopereira) [inline ably-ios/ARTAuth.m]: But it should not store all the elements and `copyWithZone:` should copy all of them as good practice.

Comment (ricardopereira) [inline ably-ios/ARTAuth.m]: [RSA10g](http://docs.ably.io/client-lib-development-guide/features/#RSA10g)

Comment (tcard) [inline ably-ios/ARTAuth.m]: Oh, fair enough.

**Comment by tcard** [comment]
LGTM

**Comment by mattheworiordan** [comment]
LGTM

**Pull request merged**